src/appshell/view/dockwindow/dockpageview.cpp

```diff
@@ -213,6 +232,130 @@ DockPanelView* DockPageView::findPanelForTab(const DockPanelView* tab) const
     return !panels.isEmpty() ? panels.first() : nullptr;
 }
 
+void DockPageView::reorderSections()
+{
+    //! NOTE: In some cases, such as setting visible true,
+    //! it is necessary to give the UI time to render the content, so we will add a delay.
+    QTimer::singleShot(2000, this, [this](){
+        doReorderSections();
+    });
+}
+
+void DockPageView::doReorderSections()
+{
+    QList<DockBase*> docks = allDocks();
+    QList<DockBase*> _docks;
+
+    for (DockBase* dock: docks) {
+        if (dock->contentNavigationPanel() && dock->isVisible()) {
+            _docks.append(dock);
+        }
+    }
+
+    QHash<ui::INavigationSection*, QList<DockBase*> > orderedSections = reorderDocksNavigationSections(_docks);
+
+    for (QList<DockBase*>& panels : orderedSections.values()) {
+        reorderNavigationSectionPanels(panels);
+    }
+}
+
+QHash<mu::ui::INavigationSection*, QList<DockBase*> > DockPageView::reorderDocksNavigationSections(QList<DockBase*>& docks)
```

Comment:
Ideally, this method should have the same interface as DockPageView::reorderNavigationSectionPanels (i.e. it returns nothing)